test/storage/bwtree_index_test.cpp

```diff
@@ -2,1269 +2,1872 @@
-          throw std::runtime_error("Invalid compact ints key type.");
-      }
-    }
+TEST_F(BwTreeIndexTests, UniqueKey4) {
+  auto *txn0 = txn_manager_.BeginTransaction();
 
-    for (uint8_t j = 0; j < 10; j++) {
-      // fill buffer pr_A with random data data_A
-      const auto data_A = FillProjectedRow(metadata, pr_A, &generator_);
-      float probabilities[] = {0.0, 0.5, 1.0};
-
-      for (float prob : probabilities) {
-        // have B copy A
-        std::memcpy(pr_B, pr_A, initializer.ProjectedRowSize());
-        auto *data_B = new byte[key_size];
-        std::memcpy(data_B, data_A, key_size);
-
-        // modify a column of B with some probability, this also updates the reference data_B
-        bool modified = ModifyRandomColumn(metadata, pr_B, data_B, prob, &generator_);
-
-        // perform the relevant checks
-        switch (key_type) {
-          case 1:
-            EXPECT_EQ(CompactIntsFromProjectedRowEq<1>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
-            EXPECT_EQ(CompactIntsFromProjectedRowCmp<1>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) < 0);
-            break;
-          case 2:
-            EXPECT_EQ(CompactIntsFromProjectedRowEq<2>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
-            EXPECT_EQ(CompactIntsFromProjectedRowCmp<2>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) < 0);
-            break;
-          case 3:
-            EXPECT_EQ(CompactIntsFromProjectedRowEq<3>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
-            EXPECT_EQ(CompactIntsFromProjectedRowCmp<3>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) < 0);
-            break;
-          case 4:
-            EXPECT_EQ(CompactIntsFromProjectedRowEq<4>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) == 0 && !modified);
-            EXPECT_EQ(CompactIntsFromProjectedRowCmp<4>(metadata, *pr_A, *pr_B),
-                      std::memcmp(data_A, data_B, key_size) < 0);
-            break;
-          default:
-            throw std::runtime_error("Invalid compact ints key type.");
-        }
-        delete[] data_B;
-      }
+  // txn 0 inserts into table
+  auto *insert_tuple = tuple_initializer_.InitializeRow(insert_buffer_);
+  *reinterpret_cast<int32_t *>(insert_tuple->AccessForceNotNull(0)) = 15721;
+  const auto tuple_slot = sql_table_->Insert(txn0, *insert_tuple);
 
-      delete[] data_A;
-    }
+  // txn 0 inserts into index
+  auto *insert_key = unique_index_->GetProjectedRowInitializer().InitializeRow(insert_buffer_);
+  *reinterpret_cast<int32_t *>(insert_key->AccessForceNotNull(0)) = 15721;
+  EXPECT_TRUE(unique_index_->InsertUnique(txn0, *insert_key, tuple_slot));
 
-    delete[]
```

Comment:
This is not an update in the table.

Reply:
See line 1197 for description. It's a Delete and Insert because it's an update on an indexed attribute.